### ISSUE TITLE
Woke fails to work with word boundary on file names

### ISSUE THREAD

**Issue by cognitivegears** [get-woke/woke#94]
Please see PR #93 for my submission of a fix for this issue. As the title says, although word boundaries are supported in options, they do not currently effect matching on filenames. Also created an issue in addition to the PR to include the detailed info below:

Thanks so much for this excellent tool! 

- [ X ] Yes, I've searched similar issues on GitHub and didn't find any.
- [ X ] Yes, I've included all information below (version, config, etc).

Please include the following information:

<details><summary>Version of woke</summary>

```console
$ woke --version
woke version 0.9.1
```

</details>

<details><summary>Config file</summary>

```console
$ cat .woke.yml
rules:
  - name: master
    terms:
      - master
    alternatives:
      - main
      - original
      - source
      - primary
      - coordinator
    options:
      word_boundary: true
  - name: slave
    terms:
      - slave
    alternatives:
      - follower
      - replica
      - standby
      - agent
      - secondary
```

</details>

<details><summary>Go environment</summary>

```console
$ go version && go env
go version go1.16.2 linux/amd64
GO111MODULE=""
GOARCH="amd64"
GOBIN=""
GOCACHE="/home/cognitivegears/.cache/go-build"
GOENV="/home/cognitivegears/.config/go/env"
GOEXE=""
GOFLAGS=""
GOHOSTARCH="amd64"
GOHOSTOS="linux"
GOINSECURE=""
GOMODCACHE="/home/cognitivegears/go/pkg/mod"
GONOPROXY=""
GONOSUMDB=""
GOOS="linux"
GOPATH="/home/cognitivegears/go"
GOPRIVATE=""
GOPROXY="https://proxy.golang.org,direct"
GOROOT="/usr/lib/go-1.16"
GOSUMDB="sum.golang.org"
GOTMPDIR=""
GOTOOLDIR="/usr/lib/go-1.16/pkg/tool/linux_amd64"
GOVCS=""
GOVERSION="go1.16.2"
GCCGO="gccgo"
AR="ar"
CC="gcc"
CXX="g++"
CGO_ENABLED="1"
GOMOD="/dev/null"
CGO_CFLAGS="-g -O2"
CGO_CPPFLAGS=""
CGO_CXXFLAGS="-g -O2"
CGO_FFLAGS="-g -O2"
CGO_LDFLAGS="-g -O2"
PKG_CONFIG="pkg-config"
GOGCCFLAGS="-fPIC -m64 -pthread -fmessage-length=0 -fdebug-prefix-map=/tmp/go-build1570773687=/tmp/go-build -gno-record-gcc-switches"
```

</details>

<details><summary>Verbose output of running</summary>

```console
❯ woke --debug master.txt
2021-07-02T14:46:19-05:00 DBG woke version 0.9.1 built from 0dc3fe3 on 2021-06-20T17:59:32Z
2021-07-02T14:46:19-05:00 DBG loaded config file config=/home/cognitivegears/.woke.yaml
2021-07-02T14:46:19-05:00 DBG rules enabled rules=["master","slave","whitelist","blacklist","master-slave","grandfathered","man-hours","sanity","dummy","guys","whitebox","blackbox"]
2021-07-02T14:46:19-05:00 DBG skipping ignorefile error="open .gitignore: no such file or directory" file=.gitignore
2021-07-02T14:46:19-05:00 DBG skipping ignorefile error="open .wokeignore: no such file or directory" file=.wokeignore
2021-07-02T14:46:19-05:00 DBG finished compiling ignores durationMS=0.10426
2021-07-02T14:46:19-05:00 DBG created new printer printer=text
2021-07-02T14:46:19-05:00 DBG process files path=master.txt type=parallel
2021-07-02T14:46:19-05:00 DBG skipping content file=master.txt reason="file is empty"
2021-07-02T14:46:19-05:00 DBG finished processing violations durationMS=0.133699 file=master.txt
master.txt:1:1-1: Filename violation: `master` may be insensitive, use `main`, `original`, `source`, `primary`, `coordinator` instead (error)
2021-07-02T14:46:19-05:00 DBG woke completed durationMS=0.662836
  ~/Documents/projects ··························································
❯ woke --debug mastertest.txt 
2021-07-02T14:46:28-05:00 DBG woke version 0.9.1 built from 0dc3fe3 on 2021-06-20T17:59:32Z
2021-07-02T14:46:28-05:00 DBG loaded config file config=/home/cognitivegears/.woke.yaml
2021-07-02T14:46:28-05:00 DBG rules enabled rules=["master","slave","whitelist","blacklist","master-slave","grandfathered","man-hours","sanity","dummy","guys","whitebox","blackbox"]
2021-07-02T14:46:28-05:00 DBG skipping ignorefile error="open .gitignore: no such file or directory" file=.gitignore
2021-07-02T14:46:28-05:00 DBG skipping ignorefile error="open .wokeignore: no such file or directory" file=.wokeignore
2021-07-02T14:46:28-05:00 DBG finished compiling ignores durationMS=0.11655
2021-07-02T14:46:28-05:00 DBG created new printer printer=text
2021-07-02T14:46:28-05:00 DBG process files path=mastertest.txt type=parallel
2021-07-02T14:46:28-05:00 DBG skipping content file=mastertest.txt reason="file is empty"
2021-07-02T14:46:28-05:00 DBG finished processing violations durationMS=0.071979 file=mastertest.txt
mastertest.txt:1:1-1: Filename violation: `mastertest` may be insensitive, use `main`, `original`, `source`, `primary`, `coordinator` instead (error)
2021-07-02T14:46:28-05:00 DBG woke completed durationMS=0.688194

```

</details>

**Comment by caitlinelfring** [comment]
Thanks for opening the issue and PR... take a look at it now

**Comment by caitlinelfring** [comment]
Released in [`v0.9.2`](https://github.com/get-woke/woke/releases/tag/v0.9.2). Thanks for the contribution!
